### PR TITLE
Use phpseclib v3

### PR DESCRIPTION
**Pull request by simPod**
I kind of cannot find out why `\League\Flysystem\PhpseclibV3\SftpConnectionProviderTest::verifying_a_fingerprint()` fails. any idea?

### CODE REVIEW

**Comment by JammingBen** [comment]
Could be because the default hash algorithm changed from `sha1` to `sha512`. Try `PublicKeyLoader::load($this->privateKey, $this->passphrase)->withHash('sha1')->withMGFHash('sha1');`

**Comment by simPod** [comment]
@JammingBen interesting but it does not even go through those lines with PublicKeyLoader.

But in SSH it detects `$server_host_key_algorithm` to be `ssh-ed25519` as `$public_key_format` is `ssh-ed25519` 🤔

**Comment by simPod** [comment]
Response with v2 is 

```...ssh-rsa����)"2:�V...```

with v3 is

```...ssh-ed25519...```

**Comment by Nyholm** [comment]
Please dont remove the source for PhpseclibV2. If you want to add support for PhpseclibV3, then create a new package.

**Comment by simPod** [comment]
@Nyholm is this what you meant?

**Comment by Nyholm** [comment]
Yes, but no files in `src/PhpseclibV3/vendor`

File: src/PhpseclibV3/composer.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "autoload": {
+        "psr-4": {
+                "League\\Flysystem\\PhpseclibV3\\": ""
+        }
+    },
+    "require": {
+        "php": "^7.2 || ^8.0",
+        "league/flysystem": "^2.0.0",
+        "league/mime-type-detection": "^1.0.0",
+        "phpseclib/phpseclib": "^2.0"
```
Comment (Nyholm) [inline src/PhpseclibV3/composer.json]: Oops

**Comment by Nyholm** [comment]
We also need to make sure all tests are green

**Comment by simPod** [comment]
That's gonna be fun. Probably will have to create custom CI steps for v2 as it seems to be exclusive with v3.

Only `composer update` is run for all packages together

**Comment by roukmoute** [comment]
Hi! 
Any news?

**Comment by simPod** [comment]
Not sure if I should go ahead with this. A lot of unrelated shuffling is required. Eg. the one with CI mentioned above.

**Pull request closed without merging**